cli/src/template_parser.rs

```diff
@@ -1149,6 +1169,12 @@ mod tests {
             parse_normalized("a ++ b"),
         );
 
+        // Not recursion because functions are overloaded by arity.
+        assert_eq!(
+            with_aliases([("F(x)", "F(x,b)"), ("F(x,y)", "x ++ y")]).parse_normalized("F(a)"),
+            parse_normalized("a ++ b")
+        );
+
```

Comment:
Since the argument names are part of `AliasId`, can we run into some funny cases with recursion here? I guess not because we look up the function to resolve to by arity and always find the same one?

Reply:
> I guess not because we look up the function to resolve to by arity and always find the same one?

Correct. There's no more than one parameter `[String]` of the same arity. It's wasting cpu time for string comparison, but I wasn't bothered to write custom Eq just for that reason.